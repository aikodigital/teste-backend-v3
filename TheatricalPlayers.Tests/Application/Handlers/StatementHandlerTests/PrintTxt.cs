using ApprovalTests;
using ApprovalTests.Reporters;
using TheatricalPlayers.Application.Handlers;
using TheatricalPlayers.Core.Enums;
using TheatricalPlayers.Core.Interfaces.Statements;
using TheatricalPlayers.Tests.Application.Mocks;
using Xunit;

namespace TheatricalPlayers.Tests.Application.Handlers.StatementHandlerTests;

public class PrintTxt
{
    private readonly IStatementPrinterHandler _statementPrinterHandler;
    public PrintTxt()
    {
        _statementPrinterHandler = new StatementPrinterHandler();
    }

    [Fact]
    [UseReporter(typeof(DiffReporter))]
    public void PrintTxtOk()
    {
        var (plays, invoice) = InvoiceSampleDataGenerator.CreateSampleData();
        
        var txt = _statementPrinterHandler.PrintTxt(invoice, plays);

        Approvals.Verify(txt);
    }
}