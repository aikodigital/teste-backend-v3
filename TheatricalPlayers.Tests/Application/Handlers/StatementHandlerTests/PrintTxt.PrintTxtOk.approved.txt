Statement for BigCo
  Hamlet: $650.00 (55 seats)
  As You Like It: $547.00 (35 seats)
  Othello: $456.00 (40 seats)
  Henry V: $705.40 (20 seats)
  john: $931.60 (39 seats)
  Richard III: $803.60 (20 seats)
Amount owed is $4,093.60
You earned 56 credits
